src/libraries/Directory.Build.targets

```diff
@@ -299,4 +299,49 @@
     </PropertyGroup>
   </Target>
 
+  <PropertyGroup>
+    <BeforePack>IncludeAnalyzersInPackage;$(BeforePack)</BeforePack>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Ensure AnalyzerReference items are restored and built 
+         The target framework of Analyzers has no relationship to that of the refrencing project, 
+         so we don't apply TargetFramework filters nor do we pass in TargetFramework.
+         When BuildProjectReferences=false we make sure to set BuildReference=false to make
+         sure not to try to call GetTargetPath in the outerbuild of the analyzer project. -->
+    <ProjectReference Include="@(AnalyzerReference)" SkipGetTargetFrameworkProperties="true" UndefineProperties="TargetFramework" ReferenceOutputAssembly="false" PrivateAssets="all" BuildReference="$(BuildProjectReferences)"  />
+  </ItemGroup>
+
+  <Target Name="IncludeAnalyzersInPackage" Condition="'@(AnalyzerReference)' != ''">
+    <!-- Call a target in the analyzer project to get all the files it would normally place in a package.
+         These will be returned as items with identity pointing to the built file, and PackagePath metadata
+         set to their location in the package.  IsSymbol metadata will be set to distinguish symbols. -->
+    <MSBuild Projects="@(AnalyzerReference)"
+             Targets="GetAnalyzerPackFiles">
+      <Output TaskParameter="TargetOutputs" ItemName="_AnalyzerFiles" />
+    </MSBuild>
```

Comment:
This breaks static graph build (which we don't use yet but better to be careful). Why can't you use a P2P with the Targets attribute instead and only use this path when building inside VS? https://github.com/dotnet/arcade/blob/44324e2d3563921f60b1522fccf3fef45dcfe636/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacadesNotSupported.targets#L21-L27

Reply:
Also, relying solely on "Targets" metadata breaks the `--no-dependencies` and `--no-build` scenario (see https://github.com/dotnet/arcade/pull/6801)

Reply:
> use a ProjectReference to invoke the GetAnalyzerPackFiles target 

This would be problematic since `ProjectReferences` are evaluated in the inner-build.  We'd be over-building by running `GetAnalyzerPackFiles` in the context of every referencing project's `TargetFrameworks`.  We'd need to deal with this by plumbing that output up to the outer build and de-duping it in the package. It's also broken in the `--no-build` and `--no-dependencies` case as I mentioned above.

> P2Ps are well understood by static graph, MSBuild calls aren't.

We [already have a P2P](https://github.com/dotnet/runtime/blob/e5b7093b4d1103f147b9a1a5ea56ee6409076353/src/libraries/Directory.Build.targets#L312) for the purposes of build and restore.  I hope this will do the job during static-graph as well.  This specific `MSBuild` invocation fetches an output group, very similar to the [other MSBuild invocations](https://github.com/NuGet/NuGet.Client/blob/87e9eab3090d2074f90d9190ecd2d95685f9e2e5/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets#L350-L417) NuGet uses to fetch package output during Pack.  It should rely on all the cached targets that happened during the main build and only run the pack-specific targets.